### PR TITLE
Fix JSON parsing errors caused by spaces and special characters in Lu…

### PR DESCRIPTION
**Pull request by emrahtokalak**
Fix JSON parsing errors caused by spaces and special characters in LuceneQuerySource

- Resolved issues with JSON parsing in `LuceneQuerySource` where spaces and special characters like `/` were causing errors.
- Updated the `parameterizedQuery` parsing logic to handle these cases correctly.

As we can see from the following Lucene query, if there are liquid expressions within the template, it causes spaces in the JSON, leading to errors.

Screenshot 2024-11-09 at 20 58 46

### CODE REVIEW

**Comment by hishamco** [comment]
Could you please add a unit test?

File: test/OrchardCore.Tests/Apis/Lucene/LuceneQueryTests.cs [hunk 1]
```diff
@@ -152,4 +152,29 @@ public async Task TwoWildcardQueriesWithBoostHasResults()
...
+        var queryResults = await content.Content.ReadAsAsync<LuceneQueryResults>();
+
+        // Test
+        Assert.NotNull(queryResults);
+        Assert.NotEmpty(queryResults.Items);
+
+        Console.WriteLine(queryResults.Items);
+        var contentItems = queryResults.Items.Select(x => JObject.FromObject(x).Deserialize<ContentItem>());
+
+        Assert.Contains("Orchard", contentItems.First().DisplayText, StringComparison.OrdinalIgnoreCase);
```
Comment (hishamco) [inline test/OrchardCore.Tests/Apis/Lucene/LuceneQueryTests.cs:178]:
```suggestion
        // Assert
        Assert.NotNull(queryResults);
        Assert.NotEmpty(queryResults.Items);

        var contentItems = queryResults.Items.Select(x => JObject.FromObject(x).Deserialize<ContentItem>());

        Assert.Contains("Orchard", contentItems.First().DisplayText, StringComparison.OrdinalIgnoreCase);
```

File: test/OrchardCore.Tests/Apis/Lucene/LuceneQueryTests.cs [hunk 2]
```diff
@@ -152,4 +152,29 @@ public async Task TwoWildcardQueriesWithBoostHasResults()
...
+    public async Task LuceneQueryTemplateWithSpecialCharactersShouldNotThrowError()
+    {
+        using var context = new LuceneContext();
+        await context.InitializeAsync();
+
+        // Act
+        // Queries containing special characters and spaces should not fail.
+        var index = "ArticleIndex";
+
+        var queryTemplate = "\r\r\n{% assign testVariable = \"48yvsghn194eft8axztaves25h\" %}\n\n{\n  \"query\": {\n    \"bool\": {\n      \"must\": [\n        { \"term\" : { \"Content.ContentItem.ContentType\" : \"Article\" } },\n        { \"term\": { \"Content.ContentItem.Published\" : \"true\" } },\n      ]\n    }\n  }\n}";
```
Comment (hishamco) [inline test/OrchardCore.Tests/Apis/Lucene/LuceneQueryTests.cs:166]:
```suggestion
        var index = "ArticleIndex";
        var queryTemplate = "\r\r\n{% assign testVariable = \"48yvsghn194eft8axztaves25h\" %}\n\n{\n  \"query\": {\n    \"bool\": {\n      \"must\": [\n        { \"term\" : { \"Content.ContentItem.ContentType\" : \"Article\" } },\n        { \"term\": { \"Content.ContentItem.Published\" : \"true\" } },\n      ]\n    }\n  }\n}";
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Review by hishamco — APPROVED** [review]

**Comment by hishamco** [comment]
@MikeAlhayek do you anything else to add here?

**Comment by MikeAlhayek** [comment]
LGTM

**Pull request merged**